### DISCUSSION TITLE
How to implement this function, similar to the YouTube homepage

### DISCUSSION THREAD

**Discussion by usekevinwu** [zauberzeug/nicegui#1559] [Q&A] [answered]
I have implemented a feature similar to the YouTube homepage, but due to my unfamiliarity with VUE, I don't know how to use nicegui to implement it. Implemented address: https://quasar.dev/layout/gallery

**Comment by rodja** [comment]
Sorry, your question is to unspecific. The link does not show a layout. Could you provide a screenshot and describe what you want to achieve in more detail?

**Reply by falkoschindler** [reply]
I fixed the link. But still, the question is totally unclear.

**Reply by usekevinwu** [reply]
Sorry for the late reply
![image](https://github.com/zauberzeug/nicegui/assets/37398179/3beeb675-c394-4f83-b659-cb4059d1d749)
Like this, I want to implement this function, such as menu, side navigation bar, search, etc。
Mainly not good at JavaScript code
thank you!

**Comment by natankeddem** [comment, marked as answer]
The first step would probably be setting up with drawers. The rest is nesting elements inside to get your desired output.
https://nicegui.io/page_layout
https://nicegui.io/documentation#page_layout

**Reply by usekevinwu** [reply]
Thank you, I'm ready to give it a try
